ivy/functional/ivy/statistical.py

```diff
@@ -1148,12 +1148,128 @@ def cumprod(
     )
 
 
-@handle_exceptions
-@handle_nestable
-@handle_array_like_without_promotion
-@handle_out_argument
+@handle_array_function
 @to_native_arrays_and_back
+@handle_out_argument
+@handle_array_like_without_promotion
+@handle_nestable
+@handle_exceptions
+def cummax(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    axis: int = 0,
+    reverse: bool = False,
+    dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    Return the cumulative minimum of the elements along a given axis.
```

Comment:
Great PR! Just a typo here and under axis, where you wrote minimum instead of maximum.